ResearchKit/Common/ORKAnswerFormat.m

```diff
@@ -2058,3 +2063,97 @@ - (BOOL)isEqual:(id)object {
 }
 
 @end
+
+#pragma mark - ORKLocationAnswerFormat
+
+@implementation ORKLocationAnswerFormat
+
+- (Class)questionResultClass {
+    return [ORKLocationQuestionResult class];
+}
+
+- (instancetype)initWithCoder:(NSCoder *)aDecoder {
+    self = [super initWithCoder:aDecoder];
+    if (self) {
+        
+    }
+    return self;
+}
+
+- (void)encodeWithCoder:(NSCoder *)aCoder {
+    [super encodeWithCoder:aCoder];
+}
+
++ (BOOL)supportsSecureCoding {
+    return YES;
+}
+
+- (void)validateParameters {
+    [super validateParameters];
+}
+
+- (BOOL)isAnswerValidWithString:(nullable NSString *)text {
+    BOOL isValid = [super isAnswerValidWithString:text];
+    
+    NSArray *components = [text componentsSeparatedByString:@", "];
+    if (components.count != 2) {
+        isValid = false;
+    } else {
+        NSNumberFormatter *decimalFormatter = [[NSNumberFormatter alloc] init];
+        decimalFormatter.numberStyle = NSNumberFormatterDecimalStyle;
+        
+        NSNumber *latitude = [decimalFormatter numberFromString:components[0]];
+        NSNumber *longitude = [decimalFormatter numberFromString:components[1]];
+        
+        if (latitude.doubleValue < -180.0 || latitude.doubleValue > 180.0) {
+            isValid = false;
+        } else if (longitude.doubleValue < -180.0 || longitude.doubleValue > 180.0) {
+            isValid = false;
+        }
```

Comment:
I don't think we need to translate the coordinates into a string and then validated them here. We can just take the coordinates and validate them in `isAnswerValid` instead of validating them in `isAnswerValidWithString:`.